src/KKS_Fix_MainGameOptimizations/AddHeroines.cs

```diff
@@ -0,0 +1,185 @@
+﻿using BepInEx;
+using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using TMPro; 
+
+namespace IllusionFixes
+{
+    public partial class MainGameOptimizations : BaseUnityPlugin
+    {
+        [HarmonyPrefix]
+        [HarmonyPatch(typeof(ActionGame.ClassRoomList), nameof(ActionGame.ClassRoomList.GetClassRegisterMax))]
+        static private bool GetClassRegisterMaxPrefix(ActionGame.ClassRoomList __instance, ref int __result, int schoolClass )
+        {
+            __result = NewGetClassRegisterMax(schoolClass);
```

Comment:
Why not add `AddHeroines.Value` to the result in a postfix instead of replacing the whole method?

Reply:
Yes, indeed it is.

Reply:
I needed to know the upper limit of the number of people from the patch function. That's why I made it a function. I forgot.